pandas/plotting/_core.py

```diff
@@ -815,11 +816,22 @@ def _post_plot_logic(self, ax, data):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, c=None, size_factor=1, **kwargs):
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
+            # Set default size if no argument is given.
             s = 20
+        elif is_hashable(s) and s in data.columns:
+            # Handle the case where s is a label of a column of the df.
+            # The data is normalized to 200 * size_factor.
+            size_data = data.loc[:, s].values
+            if is_numeric_dtype(size_data):
+                self.size_title = s
+                self.s_data_max = size_data.max()
+                self.size_factor = size_factor
+                self.bubble_points = 200
+                s = self.bubble_points * size_factor * size_data / self.s_data_max
+            else:
+                raise TypeError('s must be of numeric dtype')
```

Comment:
better error message here (s is the name of a column)

Reply:
How about "Bubbles sizes must be of numeric dtype" ?

Reply:
`'size_factor' must be numeric or categorical dtype.`